platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -184,6 +187,54 @@ class HangingProtocolService {
     this.protocols.set(protocolId, protocol);
   }
 
+  /** Sets the given protocol object as active.  Also sets the default
+   * for the given id to the specified protocol - allows updating which
+   * protocol is active for a given name.
+   */
+  public setActiveProtocol(id: string, protocol?: Protocol): void {
```

Comment:
It is weird that `setActiveProtocol` is pushing (instead of setting) into the active protocolIds list. I expected a set not append

Reply:
Will rename it.